src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx

```diff
@@ -32,13 +42,45 @@ export default class AnnotationColumnFormatter
             value = {
                 myCancerGenomeLinks: [],
                 isHotspot: false,
-                is3dHotspot: false
+                is3dHotspot: false,
+                hasCivicVariants: true
             };
         }
 
         return value;
     }
 
+   /**
+    * Returns an ICivicEntry if the civicGenes and civicVariants have information about the gene and the mutation (variant) specified. Otherwise it returns 
+    * an empty object.
+    */
+    public static getCivicEntry(copyNumberData:DiscreteCopyNumberData[], civicGenes:ICivicGene, civicVariants:ICivicVariant): ICivicEntry | null
+    {
+        let geneSymbol: string = copyNumberData[0].gene.hugoGeneSymbol;
+        let geneVariants: {[name: string]: ICivicVariantData} = civicVariants[geneSymbol];
+        let geneEntry: ICivicGeneData = civicGenes[geneSymbol];
+        let civicEntry = null;
+        //Only return data for genes with variants or it has a description provided by the Civic API
+        if (geneVariants || (geneEntry && geneEntry.description != "")) {
+            civicEntry = buildCivicEntry(geneEntry, geneVariants);
+        }
+
+        return civicEntry;
+    }
+
+    public static hasCivicVariants (copyNumberData:DiscreteCopyNumberData[], civicGenes:ICivicGene, civicVariants:ICivicVariant): boolean
+    {
+        let geneSymbol: string = copyNumberData[0].gene.hugoGeneSymbol;
+        let geneVariants: {[name: string]: ICivicVariantData} = civicVariants[geneSymbol];
+        let geneEntry: ICivicGeneData = civicGenes[geneSymbol];
+    
+        if (geneEntry && !geneVariants) {
+            return false;
+        }
+        
```

Comment:
I think this method could be more clearly written with use of hasOwnProperty method of object.   e.g.
if (!civicGenes.hasOwnProperty(geneSymbol) ...